src/d3a/models/strategy/infinite_bus.py

```diff
@@ -44,9 +44,12 @@ def __init__(self, energy_sell_rate=None, energy_rate_profile=None, energy_buy_r
         self._set_market_maker_rate()
 
     def _set_market_maker_rate(self):
-        if self.energy_rate_profile is not None:
+        if self.energy_rate_profile is not None and isinstance(self.energy_rate_profile, dict):
```

Comment:
General question: `self.energy_rate_profile` is one of the elements in the `parameters` set. Doesn't this mean that the `InfiniteBusStrategy` can't be instantiated without it?

Reply:
Its not a mandatory argument in constructor, so can be instantiated without it.